cmd/agent/install_mac_os.sh

```diff
@@ -74,6 +74,20 @@ function find_latest_patch_version_for() {
     echo "$latest_patch"
 }
 
+function prepare_dmg_file() {
+    $sudo_cmd rm -f $dmg_file
+    $sudo_cmd touch $dmg_file
+    $sudo_cmd chmod 600 $dmg_file
+
+    file_owner=$(stat -c %u $dmg_file)
+    file_permission=$(stat -c %a $dmg_file)
+    file_size=$(stat -c %s $dmg_file)
+    if [ "$file_owner" -ne 0 ] || [ "$file_permission" -ne 600 ] || [ "$file_size" -ne 0 ]; then
```

Comment:
Not sure how far we should follow [this guideline](https://google.github.io/styleguide/shellguide.html#s6.3-tests) (about using `[[..]]` instead of `[..]`)